src/array/empty.js

```diff
@@ -0,0 +1,23 @@
+define(['../lang/isArray'], function(isArray) {
+
+    /**
+     * Array empty
+     */
+
+    function empty(array) {
+        var i, len;
+
+        if(!isArray(array)) {
+            return [];
+        }
+
+        for (i = 0, len = array.length; i < len; i++) {
+            array.pop();
+        }
+
```

Comment:
It seems you are right; I was influenced by comments to http://stackoverflow.com/a/1234337 that it might not work everywhere, but i have read up on this and it seems to be OK. Would you like me to change the line to `array.length`? The performance point is still valid, but might be negligible.